chatterbot/trainers.py

```diff
@@ -102,20 +102,32 @@ def __init__(self, storage, **kwargs):
 
         self.corpus = Corpus()
 
-    def train(self, *corpora):
-        trainer = ListTrainer(self.storage)
+    def train(self, *corpus_paths):
 
         # Allow a list of corpora to be passed instead of arguments
-        if len(corpora) == 1:
-            if isinstance(corpora[0], list):
-                corpora = corpora[0]
+        if len(corpus_paths) == 1:
+            if isinstance(corpus_paths[0], list):
+                corpus_paths = corpus_paths[0]
 
         # Train the chat bot with each statement and response pair
-        for corpus in corpora:
-            corpus_data = self.corpus.load_corpus(corpus)
-            for data in corpus_data:
-                for pair in data:
-                    trainer.train(pair)
+        for corpus_path in corpus_paths:
+
```

Comment:
List comprehension is always an option to improve performance. I do have concerns about readability though as the format of list comprehension is often difficult to understand at a quick glance.